EntityLib/Exception.h

```diff
@@ -72,192 +74,193 @@ void ContextException::addContextMessage(char const* _message, Args&&... args) n
-{
-    std::exception_ptr exptr;
-    template <typename... Args>
-    WrapperException(std::exception_ptr const& _exptr, char const* _message, Args&&... _args) noexcept;
-};
-
-/// Use FileSystemError rather than filesystem_error, since
-///     - It will convert the message charset into utf8
-///     - It will print the path as expected in Wild tools
-struct FileSystemError : ContextException
-{
-    FileSystemError(
-        std::string const& _msg,
-        std::filesystem::path const& _rootPath,
-        std::filesystem::path const& _relPath,
-        std::error_code _error);
+    /// A ContextException which is wrapping an external
+    /// Usefull to add some context info on an exception which in not a ContextException
+    struct WrapperException : ContextException
+    {
+        std::exception_ptr exptr;
+        template <typename... Args>
+        WrapperException(std::exception_ptr const& _exptr, char const* _message, Args&&... _args) noexcept;
+    };
 
-    /// This ctor will use the errno as error_code
-    FileSystemError(
-        std::string const& _msg,
-        std::filesystem::path const& _rootPath,
-        std::filesystem::path const& _relPath);
-};
+    /// Use FileSystemError rather than filesystem_error, since
+    ///     - It will convert the message charset into utf8
+    ///     - It will print the path as expected in Wild tools
+    struct FileSystemError : ContextException
+    {
+        FileSystemError(
+            std::string const& _msg,
+            std::filesystem::path const& _rootPath,
+            std::filesystem::path const& _relPath,
+            std::error_code _error);
 
-// *************************************** bodies *************************************************
+        /// This ctor will use the errno as error_code
+        FileSystemError(
+            std::string const& _msg,
+            std::filesystem::path const& _rootPath,
+            std::filesystem::path const& _relPath);
+    };
 
-inline char const* formatPath(
-    std::filesystem::path const&, //_base,
-    std::filesystem::path const& _rel)
-{
-    thread_local static char buffer[2048];
-    snprintf(buffer, sizeof(buffer), R"("%s")", _rel.generic_string().c_str());
-    return buffer;
-}
+    // *************************************** bodies *************************************************
 
-inline ToolsException::ToolsException(char const* _message) noexcept
-    : std::runtime_error(_message == nullptr ? "" : _message)
-{
-}
+    inline char const* formatPath(
+        std::filesystem::path const&, //_base,
+        std::filesystem::path const& _rel)
+    {
+        thread_local static char buffer[2048];
+        snprintf(buffer, sizeof(buffer), R"("%s")", _rel.generic_string().c_str());
+        return buffer;
+    }
 
-inline ContextException::ContextException() noexcept = default;
+    inline ToolsException::ToolsException(char const* _message) noexcept
+        : std::runtime_error(_message)
```

Comment:
The previous code was lost in the merge.
```suggestion
        : std::runtime_error(_message == nullptr ? "" : _message)
```